### PR TITLE
IPC Overloaders

### PR DESCRIPTION
**Pull request by Geevies**
* Added an IPC overloader vendor to the bar lounge. Overloaders are legal luxury items that contain software that act like drugs for IPCs.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by DreamySkrell** [comment]
Suggestion: Please add these to random stuff or whatever, so they spawn randomly in the warehouse or on third party ships

**Review by niennab — APPROVED** [review]

File: code/game/objects/items/ipc_overloaders.dm [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+/obj/item/ipc_overloader/proc/handle_overdose_effect(var/mob/living/carbon/human/target)
+	switch(rand(1, 8))
+		if(1) // random limb fire damage
+			to_chat(target, SPAN_WARNING("Your limbs rapidly heat up!"))
+			target.take_overall_damage(0, 20)
+		if(2) // random component damage
+			to_chat(target, SPAN_WARNING("Your internals rapidly heat up!"))
+			var/probability = 100
+			for(var/obj/item/organ/internal/internal_component in shuffle(target.internal_organs))
```
Comment (mikomyazaki) [inline code/game/objects/items/ipc_overloaders.dm:92]:
```suggestion
			for(var/obj/item/organ/internal/internal_component as anything in shuffle(target.internal_organs))
```

This list only contains stuff of this type right? Or does it contain stuff like implants too, I think it doesn't.

File: code/game/objects/items/ipc_overloaders.dm [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+	. = ..()
+	to_chat(target, SPAN_NOTICE("You feel alert."))
+	target.dizziness = 0
+	target.make_dizzy(200)
+
+/obj/item/ipc_overloader/screenshaker/midway_overloader_effect(mob/living/carbon/human/target)
+	. = ..()
+	if(.)
+		to_chat(target, SPAN_NOTICE("You feel alert."))
+		target.dizziness = 0
```
Comment (mikomyazaki) [inline code/game/objects/items/ipc_overloaders.dm:222]: The make_dizzy proc already has a cap of 1000, do we need to set a cap for this drug specifically of 200? Is it like a lore/IC reason that plugging in two overloaders doesn't equal the effect of two overloaders?

**Review by alsoandanswer — APPROVED** [review]

File: code/game/objects/items/ipc_overloaders.dm (hunk 2 above)
Comment (Geevies) [reply code/game/objects/items/ipc_overloaders.dm:222]: at the moment it's pretty hard to determine if it's a midway effect or multiple overloaders being used, and the idea of this one is that it's just a gentle screenshake, unlike the 1000 which goes ballistic, so i think for this it's fine with the 200 cap

**Comment by Geevies** [comment]
!review

**Pull request merged**